### DISCUSSION TITLE
Getting <UKN><UKN><UKN><UKN> as prediction from custom trained model

### DISCUSSION THREAD

**Discussion by gastodler** [open-mmlab/mmocr#1939] [Q&A]
Hi, 

I trained a model on my Arabic data set (using SAR). 
When I tried to inference on some image using my checkpoint (after 10 epochs - just to test it) I got the _UKN_ predicition (probably unknown). 

Does anyone knows what could be the reason for that? 

Thank you in advance

**Comment by gaotongxiao** [comment]
Probably you haven't modified the dictionary yet to incorporate Arabic characters, leading to all of them being recognized as unknown tokens UKN. Please refer to https://mmocr.readthedocs.io/en/dev-1.x/get_started/faq.html#text-recognition for instruction and try again

**Reply by gastodler** [reply]
@gaotongxiao Thanks for your reply. 

1. I have changed the dict as you said. 
2. I can also see it in the cfg file that is created during the training:

`dictionary = dict(
    type='Dictionary',
    dict_file=
    '/home/ubuntu/clipper/lib/python3.10/site-packages/mmocr/.mim/dicts/arabic_digits_symbols.txt'`

3. Here is a sample of my dict:

`
ﺍ
ﺏ
ﺕ
ﺙ
ﺝ
ﺡ
ﺥ
ﺩ`

Can you think of some other reason? 

Thank you in advance.

**Reply by gastodler** [reply]
Hi, @gaotongxiao 
I was able to get Arabic letters in my inference pipeline. 
It seems that the dict needs to be created using the font I created my labels (in my case _Amiri-Regular_). 

If you have something to add about it, I'll be happy to hear.
Thanks for your help :)
